card_ui/card_states/card_released_state.gd

```diff
@@ -0,0 +1,16 @@
+extends CardState
+
+var played: bool
+
+func enter() -> void:
+	card_ui.state.text = "PLAYED"
+	played = false
+	
+	if not card_ui.targets.is_empty():
+		played= true
+		print("play card for target", card_ui.targets)
+	
+func on_input(_event: InputEvent) -> void:
+	if played:
```

Comment:
Is there a reason why the code at lines 14 and 15 are here? It seems like to me these lines are responsible for having the card freeze once played on the board. However, if you remove these lines and drop the card on the board then according to the printed output the card is still played and the card returns back to the deck rather than returning to your hand, which I think would be better behavior since you would want to have the card available again the next turn. I would like these lines to be removed or provided some explanation for keeping these lines (ex. like if removing them would introduce a bug).

Reply:
I went and added comments to my code. Also, I removed the cards freezing and just let them go back to base state after being played for now. I was just unsure of if the cards were replayable or single-use.